src/engine/video/gl/gl_render_target.cpp

```diff
@@ -0,0 +1,292 @@
+////////////////////////////////////////////////////////////////////////////////
+//            Copyright (C) 2012-2015 by Bertram (Valyria Tear)
+//                         All Rights Reserved
+//
+// This code is licensed under the GNU GPL version 2. It is free software
+// and you may modify it and/or redistribute it under the terms of this license.
+// See http://www.gnu.org/copyleft/gpl.html for details.
+////////////////////////////////////////////////////////////////////////////////
+
+/** ****************************************************************************
+*** \file    gl_render_target.cpp
+*** \author  Authenticate, James Lammlein
+*** \brief   Source file for buffers for a render target.
+*** ***************************************************************************/
+
+#include "utils/utils_pch.h"
+#include "gl_render_target.h"
+
+#include "utils/exception.h"
+#include "utils/utils_strings.h"
+
+namespace vt_video
+{
+namespace gl
+{
+
+RenderTarget::RenderTarget(unsigned width,
+                           unsigned height) :
+    _width(width),
+    _height(height),
+    _framebuffer(0),
+    _texture(0),
+    _renderbuffer_depth(0)
+{
+    assert(_width > 0);
+    assert(_height > 0);
+
+    bool errors = false;
```

Comment:
Yes, I agree.

So, I always try to write functions which never exit early.  They run to completion every time and do not have cases which cause them to short circuit and exit early.

I believe code written in this manner is easier to read and to understand.  (This is debatable and subjective.  It could be totally wrong.) :)

That is why it is written the way it is.  Now, that does not mean it is great or perfect or does not need changed.  However, that is it at least the methodology of what I was trying to achieve.

So, the code should proceed from top to bottom and do basically one thing per 'if' block.  If something goes wrong, the code should log the problem, update the flag, and proceed downwards to the next 'if' block.  However, in this case, it will skip over any remaining tasks and eventually exit.

I think one of the big weakness, which you pointed out, is the caller never knows about any problems (other than the log messages).  So, the code always succeeds, even when it does not actually or fully succeed.